packages/SwingSet/src/vats/vattp/vat-vattp.js

```diff
@@ -16,181 +23,253 @@ import { Far } from '@endo/marshal';
-
-  const makeConnectionName = makeCounter(n => `connection-${n}`);
-
-  /*
-  // A:
-  E(agoric.ibcport[0]).addListener( { onAccept() => {
-    const { host, handler } = await E(agoric.vattp).makeNetworkHost('ag-chain-B', comms);
-    const helloAddress = await E(host).publish(hello);
-    // helloAddress = '/alleged-chain/${chainID}/egress/${clistIndex}'
-    return handler;
-   });
-
-   // B:
-   const { host, handler } = await E(agoric.vattp).makeNetworkHost('ag-chain-A', comms, console);
-   E(agoric.ibcport[1]).connect('/ibc-port/portADDR/ordered/vattp-1', handler);
-   E(E(host).lookup(helloAddress)).hello()
-  */
-
-  function makeNetworkHost(allegedName, comms, cons = console) {
-    const makeAddress = makeCounter(
-      n => `/alleged-name/${allegedName}/egress/${n}`,
-    );
-    const exportedObjects = new Map(); // address -> object
-    const locatorUnum = harden({
-      lookup(address) {
-        return exportedObjects.get(address);
-      },
-    });
-    const { promise: theirLocatorUnum, resolve: gotTheirLocatorUnum } =
-      makePromiseKit();
-    const name = makeConnectionName();
-    let openCalled = false;
-    assert(!connectionNames.has(name), X`already have host for ${name}`);
-    const host = Far('host', {
-      publish(obj) {
-        const address = makeAddress();
-        exportedObjects.set(address, obj);
-        return address;
-      },
-      lookup(address) {
-        return E(theirLocatorUnum).lookup(address);
-      },
-    });
-
-    const handler = Far('host handler', {
-      async onOpen(connection, ..._args) {
-        // make a new Remote for this new connection
-        assert(!openCalled, X`host ${name} already opened`);
-        openCalled = true;
-
-        // transmitter
-        const transmitter = harden({
-          transmit(msg) {
-            // 'msg' will be a string (vats/comms/outbound.js deliverToRemote)
-            E(connection).send(msg);
-          },
-        });
-        // the comms 'addRemote' API is kind of weird because it's written to
-        // deal with cycles, where vattp has a tx/rx pair that need to be
-        // wired to comms's tx/rx pair. Somebody has to go first, so there's
-        // a cycle. TODO: maybe change comms to be easier
-        const receiverReceiver = harden({
-          setReceiver(receiver) {
-            receivers.set(connection, receiver);
-          },
-        });
-        await E(comms).addRemote(name, transmitter, receiverReceiver);
-        await E(comms).addEgress(name, 0, locatorUnum);
-        gotTheirLocatorUnum(E(comms).addIngress(name, 0));
-      },
-      onReceive(connection, msg) {
-        // setReceiver ought to be called before there's any chance of
-        // onReceive being called
-        E(receivers.get(connection)).receive(msg);
+  // QUESTION: Is there a convention around capitalization here?
+  const mailboxKindHandle = provideKindHandle(baggage, 'mailbox');
```

Comment:
e.g., "mailbox" vs. "Mailbox"

Reply:
I've been using `mailboxHandle`. My vat-timer work has a batch of handle creations at the beginning:
```js
  // first, create the KindHandles for all our durable objects
  const serviceHandle = provideKindHandle(baggage, 'timerServiceHandle');
  const oneShotEventHandle = provideKindHandle(baggage, 'oneShotEventHandle');
  const promiseEventHandle = provideKindHandle(baggage, 'promiseEventHandle');
  const repeaterEventHandle = provideKindHandle(baggage, 'repeaterEventHandle');
  const repeaterHandle = provideKindHandle(baggage, 'repeater');
  const notifierHandle = provideKindHandle(baggage, 'notifier');
```

huh, which isn't consistent

It doesn't matter too much, but now I'm thinking I want e.g. `mailboxKindHandle`, so the name of the baggage key matches the name of the variable that's holding the value. I'll change mine to folow that rule.

Reply:
`mailboxHandle = provideKindHandle(baggage, 'mailboxHandle')` reads strangely to me... if I understand correctly, "mailboxHandle" maps to a key in baggage whose associated value is meaningful only for uniquely identifying the mailbox object kind which has its behavior defined (in each version) by `defineDurableKindMulti(mailboxHandle, …)`.

So would something like `mailboxKind = provideKindHandle(baggage, 'mailboxKind')` be a more accurate representation of the underlying mechanics?